komfydb/storage/buffer_pool.cc

```diff
@@ -105,4 +105,20 @@ absl::Status BufferPool::InsertTuples(
   return absl::OkStatus();
 }
 
+absl::Status BufferPool::RemoveTuples(std::vector<RecordId>& ids,
+                                      uint32_t table_id, TransactionId tid) {
+  ASSIGN_OR_RETURN(DbFile * dbfile, catalog->GetDatabaseFile(table_id));
+  int page_count = dbfile->PageCount(), all_tuples = ids.size();
+  Page* page = nullptr;
+  for (int i = 0; i < all_tuples; i++) {
+    if (!page || ids[i].GetPageId() != page->GetId()) {
+      ASSIGN_OR_RETURN(
+          page, GetPage(tid, ids[i].GetPageId(), Permissions::READ_WRITE));
+    }
+    RETURN_IF_ERROR(page->RemoveRecord(ids[i]));
+    page->SetDirty(true, tid);
+  }
+  return absl::OkStatus();
+}
+
```

Comment:
Id give more descriptive variable names, ids_to_remove (im not convinced to this one either)